src/relay/op/tensor/transform.cc

```diff
@@ -1891,81 +1952,163 @@ Array<Array<Layout> > StridedSliceInferCorrectLayout(
+    auto strides_ndarray = runtime::NDArray::Empty({int64_t(new_begin.size())},
+                                                   DataType::Int(64), ctx);
+    int64_t* begin_data = static_cast<int64_t*>(begin_ndarray->data);
+    int64_t* end_data = static_cast<int64_t*>(end_ndarray->data);
+    for (size_t i = 0; i < new_begin.size(); ++i) {
+      begin_data[i] = new_begin[i];
+      end_data[i] = new_end[i];
+    }
+    params->begin = Constant(begin_ndarray);
+    params->end = Constant(end_ndarray);
+  }
+  return {{layout, Layout("C"), Layout("C"), Layout("C")}, {layout}};
+}
+
+inline te::Tensor DynamicStridedSlice(const te::Tensor& input,
+                                  const te::Tensor& begin,
+                                  const te::Tensor& end,
+                                  const te::Tensor& strides,
+                                  std::string name = "T_strided_slice_dynamic",
+                                  std::string tag = topi::kInjective) {
+  int64_t src_tensor_dim = input->shape.size();
+  Array<IndexExpr> out_shape;
+  for (int64_t i = 0; i < src_tensor_dim; ++i) {
+    out_shape.push_back(tvm::tir::Var("dim"));
+  }
+  // TODO(yongwww): move the compute into topi
+  return te::compute(out_shape, [&](const Array<tvm::tir::Var>& indices) {
+      Array<IndexExpr> real_indices;
+      for (int32_t i = 0; i < src_tensor_dim; ++i) {
+        real_indices.push_back(indices[i] * strides(i) + begin(i));
+      }
+      return input(real_indices);
+    }, name, tag);
 }
 
 Array<te::Tensor> StridedSliceCompute(const Attrs& attrs,
                                       const Array<te::Tensor>& inputs,
                                       const Type& out_type) {
   const StridedSliceAttrs *param = attrs.as<StridedSliceAttrs>();
   CHECK(param != nullptr);
-  return Array<te::Tensor>{
-    topi::strided_slice(inputs[0], param->begin, param->end, param->strides)
-  };
+  const ConstantNode *cbegin, *cend, *cstrides;
+  if ((cbegin = param->begin.as<ConstantNode>()) &&
+      (cend = param->end.as<ConstantNode>()) &&
+      (cstrides = param->strides.as<ConstantNode>())) {
+    Array<Integer> begin, end, strides;
+    int64_t* strides_val = ToVector(cstrides->data);
+    for (int64_t i = 0; i < cstrides->data.Shape().front(); ++i) {
+      strides.push_back(strides_val[i]);
+    }
+    int64_t* begin_val = ToVector(cbegin->data);
+    for (int64_t i = 0; i < cbegin->data.Shape().front(); ++i) {
+      begin.push_back(begin_val[i]);
+    }
+    int64_t* end_val = ToVector(cend->data);
+    for (int64_t i = 0; i < cend->data.Shape().front(); ++i) {
+      end.push_back(end_val[i]);
+    }
+    return Array<te::Tensor>{
+      topi::strided_slice(inputs[0], begin, end, strides)
+    };
+  } else {
+    te::Tensor data = inputs[0];
+    te::Tensor begin = inputs[1];
+    te::Tensor end = inputs[2];
+    te::Tensor strides = inputs[3];
+    // Dynamic computation
```

Comment:
We might want to enforce user to provide full begin, end and strides for symbolic attr case, since dealing with these inside topi would be not ideal.